### ISSUE TITLE
Parametric EQ ?

### ISSUE THREAD

**Issue by Mrnofish** [ThePBone/RootlessJamesDSP#129]
Let me apologize if this has already been discussed elsewhere, but I could find no mention of it.

Would it be possible to have a Parametric EQ implemented?

The Arbitrary Response Equalizer and AutoEQ are cool, of course.

However, I very much prefer using oratory1990's hand-tweaked presets, which are constructed for a 10-band Parametric EQ, whereas an AutoEQ profile has dozens and dozens of nodes.

The Parametric EQ is much better because it allows further, easy customization of the sound directly on the device.

To illustrate why, here is an excerpt from one of oratory1990's documents:

```
Adjust gain of band 1 to preference (Bass)
Adjust gain of band 5 to preference (boxiness/tonality)
Adjust gain of band 8 to preference (treble presence)
Adjust gain of band 10 to preference (airiness)
```

In this case two of the bands are shelf, and two are peak with specific Q-factors. With just four sliders, it is possible to affect wide ranges of frequencies easily, and do so in a way that is meaningful and easily comprehensible for everyone.

As far as I understand Wavelet has a Parametric EQ, and that's the reason why the AutoEQ project maintainer recommends it.

However, RootlessJamesDSP works better on my device, and of course I would rather use a FOSS project over a closed-source one whenever possible.

**Comment by Mrnofish** [comment]
Having to load a file produced by an external tool is completely different process than using sliders, which can be altered in real-time, unlike having to modify the file and then load it again, a very clunky workflow made more difficult by the fact that the tool, as you have pointed out, is a discontinued, limited feature-set project, additionally it's only available on Google Play, unlike RJDSP itself.

I don't mean any disrespect there, just pointing out why that VDC can only be considered a distant approximation and poor substitute of the real thing.

It does seem to me, too, that the technology underpinning a Parametric EQ feature is already in RJDSP, however the UI is a crucial part of the package, not a frill.

So in that light, I was hoping putting together the UI would not be a huge undertaking, but admittedly I can't do that myself and perhaps I am underestimating the amount of work that would need to go into it.

I had already checked the original project before opening a ticket here, that one seems a complete non-starter, though.

**Comment by ThePBone** [comment]
Building a custom UI for parametric EQs that 'feels good' is quite hard, at least on mobile devices with small touchscreens. For the ddctoolbox app, users would just input all the values by hand and could then look at a read-only preview of the frequency response.
I could have just copy-and-pasted that app into RJDSP, but I wasn't happy with its approach and rather continued the 'tradition' of keeping it simple and using .vdc-files just as Viper4Android and official JDSP UI does, which is also the intended use case by the original devs.
Most users are using the Graphic EQ or FIR EQ instead, so there wasn't any demand for an additional UI for the parametric EQ disguised as ViPER-DDC until now.
Anyway, this kind of feature would be indeed nice to have, but I don't have the time or motivation to work on it at the moment (or this app in general). I'll try to return to this later when I have more spare time, but I can't provide any ETA for this.

**Comment by Mrnofish** [comment]
The interest in such a feature might be greater than it appears at first glance, in fact there is a similar inquiry that dates back to 2018 for JDSP, and in relevant online communities, there seems to be a great deal of fondness for system-wide EQ apps, also because supplemental EQ'ing hardware is relatively costly and not much fun to carry around. Wavelet currently stands almost alone in that niche.

Perhaps arranging a bounty or something along the lines could provide some motivation.

**Comment by donjoe0** [comment]
FWIW there's another route involving a not-discontinued and not-limited-feature 3rd-party tool, but still requiring that you generate a specific file to then copy to your phone: use Audacity or your preferred waveform editor to apply whatever EQ profile you want to an initial raw/WAV containing a single 0 dBFS impulse, then rename that to .irs and use it with the Convolver function. (Hope I'm not getting this wrong and the JDSP Convolver works with raw/WAV/irs just like V4A. Can't check for myself yet as RLJDSP doesn't have an Install button when I open it in my F-Droid for some reason.)

**Comment by RHimadiiev** [comment]
Yes, In need it too. And also because of oratory1990 presets and possibility to tweak it in real time.
